doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+        close( unit )
+        open( newunit=unit, file='test.txt', status='old', &
+            form='formatted', action='read' )
+        call set3 % read_bitset(unit, advance='no')
+        call set4 % read_bitset(unit, advance='no')
+        call set5 % read_bitset(unit)
+        if ( set3 == set0 .and. set4 == set1 .and. set5 == set2 ) then
+            write(*,*) WRITE_BITSET to READ_BITSET through unit worked.'
+        end if
+    end program demo_write_bitset
+```
+
+### `xor` - bitwise exclusive `or`
+
+#### Status
+
+Experimental
+
+#### Description
+
+Replaces `set1`'s bitset with the bitwise exclusive `or` of the
+original bits of `set1` and `set2`. Note `set1` and `set2` must have
+the samee number of bits, otherwise the result is undefined.
+
+#### Syntax
+
+`result = [[stdlib_bitsets(module):xor(interface)]] (set1, set2)`
+
+#### Class
+
+Elemental subroutine
+
+#### Arguments
+
+`set1`: shall be a scalar `bitset_64` or `bitset_large` variable. It
+is an `intent(inout)` argument. On return the values of the bits in
+`set1` are the bitwise exclusive `or` of the original bits in `set1`
+with the corresponding bits in `set2`.
+
+`set2` shall be a scalar expression of the same type as `set1`. It is
+  an `intent(in)` argument. Note `set1` and `set2` must have the
+samee number of bits, otherwise the result is undefined.
+
+#### Example
+
+```fortran
+    program demo_xor
+        use stdlib_bitsets
+        type(bitset_large) :: set0, set1
+        call set0 % init(166)
+        call set1 % init(166)
+        call xor( set0, set1 ) ! none none
+        if ( none(set0) ) write(*,*) 'First test of XOR worked.'
+        call set0 % not()
+        call xor( set0, set1 ) ! all none
+        if ( all(set0) ) write(*,*) 'Second test of XOR worked.'
+        call set0 % not()
+        call set1 % not()
+        call xor( set0, set1 ) ! none all
+        if ( all(set0) ) write(*,*) 'Third test of XOR worked.'
+        call set0 % not()
+        call xor( set0, set1 ) ! all all
+        if ( none(set0) ) write(*,*) 'Fourth test of XOR worked.'
+    end program demo_xor
+```
+
+## Specification of the `stdlib_bitsets` operators
+
+### `==` - compare two bitsets to determine whether the bits have the same value
+
+#### Status
+
+Experimental
+
+#### Description
+
+Returns `.true.` if all bits in `set1` and `set2` have the same value,
+`.false.`  otherwise.
+
+#### Syntax
+
+`Result = set1 [[stdlib_bitsets(module):==(interface)]] set2
+
+#### Class
+
+Elemental operator
+
+#### Arguments
+
+`set1`: shall be a scalar `bitset_64` or `bitset_large` expression. It
+is an `intent(in)` argument.
+
+`set2`: shall be a scalar expression of the same type as `self`. It
+will have the same number of bits as `set1`. It is an `intent(in)`
+argument.
+
+#### Result character
+
+The result is a default logical scalar.
+
+#### Result value
+
+The result is `.true.` if the bits in both bitsets are set
+  to the same value, otherwise the result is `.FALSE.`.
```

Comment:
```suggestion
#### Result value

The result is a default logical scalar.
The result is `.true.` if the bits in both bitsets are set
  to the same value, otherwise the result is `.false.`.
```